SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp

```diff
@@ -0,0 +1,271 @@
+    esimd::simd<DataT, NumElems> instance(acc, offset, alignment);
+    instance.copy_to(out);
+  }
+};
+
+// Descriptor class for constructor call within r-value in an expression context
+struct rval_in_expr {
+  static std::string get_description() { return "rvalue in an expression"; }
+
+  template <typename DataT, int NumElems, typename AccessorT,
+            typename AlignmentT>
+  static void call_simd_ctor(DataT *const out, const AccessorT &acc,
+                             unsigned int offset, AlignmentT alignment) {
+    esimd::simd<DataT, NumElems> instance;
+    instance = esimd::simd<DataT, NumElems>(acc, offset, alignment);
+    instance.copy_to(out);
+  }
+};
+
+// Descriptor class for constructor call within const reference context
+class const_ref {
+public:
+  static std::string get_description() { return "const reference"; }
+
+  template <typename DataT, int NumElems, typename AccessorT,
+            typename AlignmentT>
+  static void call_simd_ctor(DataT *const out, const AccessorT &acc,
+                             unsigned int offset, AlignmentT alignment) {
+    call_simd_by_const_ref<DataT, NumElems>(
+        esimd::simd<DataT, NumElems>(acc, offset, alignment), out);
+  }
+
+private:
+  template <typename DataT, int NumElems>
+  static void
+  call_simd_by_const_ref(const esimd::simd<DataT, NumElems> &simd_by_const_ref,
+                         DataT *const out) {
+    simd_by_const_ref.copy_to(out);
+  }
+};
+
+// Test case description for logging purposes
+template <int VecSize, typename ContextT>
+struct LoadCtorAccTestDescription : public ITestDescription {
+  using BaseT = ctors::LoadCtorTestDescription<VecSize, ContextT>;
+
+public:
+  LoadCtorAccTestDescription(const std::string &vec_data_name, int dims,
+                             const std::string &acc_mode_name,
+                             const std::string &acc_target_name,
+                             unsigned int offset,
+                             const std::string &alignment_name) {
+    std::ostringstream stream;
+
+    using BaseDescriptionT = ctors::LoadCtorTestDescription<VecSize, ContextT>;
+    const BaseDescriptionT base_description(vec_data_name, alignment_name);
+    const AccessorDescription accessor_description(
+        vec_data_name, dims, acc_mode_name, acc_target_name);
+
+    stream << base_description.to_string();
+    stream << ", with offset " << log::stringify(offset);
+    stream << ", from " << accessor_description.to_string();
+
+    m_description = stream.str();
+  }
+
+  std::string to_string() const override { return m_description; }
+
+private:
+  std::string m_description;
+};
+
+// The main test routine
+//
+template <typename DataT, typename VecSizeT, typename AccDimsT,
+          typename AccModeT, typename AccTargetT, typename ContextT,
+          typename OffsetGeneratorT, typename AlignmentT>
+class run_test {
+  static constexpr int VecSize = VecSizeT::value;
```

Comment:
```suggestion
  static constexpr int NumElems= NumElemsT::value;
```
What do you think about this?
`NumElems` used in all other tests

Reply:
Fair, will fix ASAP

Reply:
Or you could leave current variant to be more specific that this variable related to `simd` size